workspaces/ui-v2/src/lib/__tests/diff-helpers/fixture.ts

```diff
@@ -32,6 +32,14 @@ export const testCase = (basePath: string) => async (
     require(path.join(__dirname + '/universes', basePath, name + '.json'))
   );
 
+  // Overwrite the learn_undocumented_bodies to ensure deterministic id generation
```

Comment:
cool, i'm fine with this, but ideally we'd "just" pass the id generation strategy from the entry point i.e. when creating an inmemory opticcontext

Reply:
the entry point should be trusted to set the right environment for clocks, ids, etc.